@hiroshi-matsuda-rit 

I tried to run the script for some models, but I could not successfully convert to hf models because of `len(tokenizer) == model_config["vocab_size"]` assertion's failure.

For example:
<details>
<summary>
exp9 model and ver3.0b1 tokenizer
</summary>

```
Phase 5: Update model config with Hugging Face Tokenizer special token settings
python convert_special_token_settings.py /model/7B_HF_RE/test_conversion_script/
Traceback (most recent call last):
  File "/model/sosuke/modelwg/convert_special_token_settings.py", line 38, in <module>
    main()
  File "/model/sosuke/modelwg/convert_special_token_settings.py", line 32, in main
    new_model_config = update_special_token_settings(tokenizer, model_config)
  File "/model/sosuke/modelwg/convert_special_token_settings.py", line 8, in update_special_token_settings
    assert len(tokenizer) == model_config["vocab_size"], f'Different vocab size definitions: {len(tokenizer)=} and {model_config["vocab_size"]=}'
AssertionError: Different vocab size definitions: len(tokenizer)=99574 and model_config["vocab_size"]=99584
```
</details>

<details>
<summary>
exp4+ model and ver2.2 50k tokenizer
</summary>

```
==========================================
Phase 5: Update model config with Hugging Face Tokenizer special token settings
python convert_special_token_settings.py /model/7B_HF_RE/test_conversion_script_ver2.2_50k/
Traceback (most recent call last):
  File "/model/sosuke/modelwg/convert_special_token_settings.py", line 38, in <module>
    main()
  File "/model/sosuke/modelwg/convert_special_token_settings.py", line 32, in main
    new_model_config = update_special_token_settings(tokenizer, model_config)
  File "/model/sosuke/modelwg/convert_special_token_settings.py", line 8, in update_special_token_settings
    assert len(tokenizer) == model_config["vocab_size"], f'Different vocab size definitions: {len(tokenizer)=} and {model_config["vocab_size"]=}'
AssertionError: Different vocab size definitions: len(tokenizer)=48586 and model_config["vocab_size"]=48640
```
</details>

While reading [this issue](https://github.com/huggingface/transformers/issues/4875) in huggingface transformers repo, I think tokenizer's vocab size and model's vocab size can be different to make model's vocab size GPU-friendly.
(I have not looked into Megatron-LM's logic to determine `vocab_size` in model_config.json)

In fact, the values of vocab_size in model_config.json, such as 48640 (= 2^9 * 5 * 19) and 99584 (= 2^8 * 389), can be factored into products of large powers of two and a few prime numbers.

I think this assertion should be removed.

--- a/convert_special_token_settings.py
+++ b/convert_special_token_settings.py
@@ -5,7 +5,8 @@ from transformers import AutoTokenizer
 
 
 def update_special_token_settings(tokenizer, model_config: dict):
-    assert len(tokenizer) == model_config["vocab_size"], f'Different vocab size definitions: {len(tokenizer)=} and {model_config["vocab_size"]=}'
+    if len(tokenizer) != model_config["vocab_size"]:
+        print(f'Might need checking: different vocab size definitions: {len(tokenizer)=} and {model_config["vocab_size"]=}', file=sys.stderr)
     new_model_config = dict(model_config)
     for token, id in [
         ("bos_token_id", tokenizer.bos_token_id),

